rocAL/rocAL/source/pipeline/ring_buffer.cpp

```diff
@@ -380,15 +415,18 @@ RingBuffer::~RingBuffer()
                 if (_host_sub_buffers[buffIdx][sub_buf_idx])
                     free(_host_sub_buffers[buffIdx][sub_buf_idx]);
             }
-            if(_host_meta_data_buffers.size() != 0) {
-                for (unsigned sub_buf_idx = 0; sub_buf_idx < _host_meta_data_buffers[buffIdx].size(); sub_buf_idx++) {
-                    if (_host_meta_data_buffers[buffIdx][sub_buf_idx])
-                        free(_host_meta_data_buffers[buffIdx][sub_buf_idx]);
-                }   
+        if(_host_meta_data_buffers.size() != 0) {
+            for (unsigned sub_buf_idx = 0; sub_buf_idx < _host_meta_data_buffers[buffIdx].size(); sub_buf_idx++) {
+                if (_host_meta_data_buffers[buffIdx][sub_buf_idx])
+                    free(_host_meta_data_buffers[buffIdx][sub_buf_idx]);
+                if (_host_roi_buffers[buffIdx][sub_buf_idx])
```

Comment:
We need to add a check if host roi buffers are enabled line line 418

Reply:
Added